app/[[...path]]/layout.tsx

```diff
@@ -0,0 +1,20 @@
+import type { ReactNode } from 'react';
+import { createElement } from 'react';
+
+import { Layout as RootLayout } from '@/theme/_layout';
+import router from '@/theme/_router';
+
+export default async function Layout(props: { children: ReactNode; params: { path?: string[] } }) {
+    const { params, children } = props;
+    const { path = [] } = params;
+
+    const { match, Layout: PageLayout } = await router(path);
+
+    if (PageLayout) {
+        // @ts-expect-error Server Component
```

Comment:
This is a known issue of Typescript JSX declarations not being (yet) compatible with async ServerComponents model.

The current workaround is to silence these errors with a `@ts` directive:
https://github.com/vercel/next.js/issues/42292#issuecomment-1298483608

Reply:
Oh, actually, it seems the newest TS package may be fine with it. I'll try it later.

Reply:
Yeah, I fixed this error in the website after upgrading to latest TS

Reply:
fb051caebee910c2e9f70fb1f1530ef857cd304d